LayoutTests/webanimations/document-timeline-current-time-updates-without-running-animations.html

```diff
@@ -0,0 +1,32 @@
+<html>
+<head>
+    <title>The document timeline's current time should increase when the page rendering does not update.</title>
+    <script src="../resources/testharness.js"></script>
+    <script src="../resources/testharnessreport.js"></script>
+</head>
+
+<body>
+
+<script>
+
+const timeout = interval => new Promise(resolve => setTimeout(resolve, interval));
+
+promise_test(async test => {
+    // Wait for a page rendering update and save the timeline's time.
+    await new Promise(requestAnimationFrame);
+    const initialCurrentTime = document.timeline.currentTime;
+
+    // Now wait a beat to check that this time is still current in the next run loop.
+    await timeout(0);
+    assert_equals(document.timeline.currentTime, initialCurrentTime);
+
+    // Now wait longer than it should take for a page rendering update to last
+    // and check that the current time has changed to a larger value.
+    await timeout(20);
```

Comment:
I think might be flakey on loaded machines. Do you have to wait but explicitly not run `Page::updateRendering`? There's normally a flurry of those around page load that are hard to avoid, so if you really want to wait for idle, you may need to wait for 4-5 updateRenderings before starting the test.

Reply:
This test reliably fails for me prior to this patch. I also expect `setTimeout(…, 20)` to always run after a WebCore `Timer` set for `16_ms` if both are generally started in the same run loop.